Refactored + some bugs fixed + new admin_bar feature

pryley opened the pull request.

Commit:
- refactored class

Commit:
Further refactoring

- convert camelCase to underscores
- fixed save to database
- fixed $this->options['auto_create'] verification

Commit:
added a link to pydio in the admin_bar

- added a link to pydio in the admin_bar
- changed the access modifier of self::set_glue_globals() to private

Commit:
Separate capability filter hooks

cfoellmann commented:
thanks for the input. I have a few concerns:
1) refactoring is completely against WordPress coding standards/code styling
2) refactoring removes DocBlocks which is necessary for code quality and documentation generation

Would you like to keep the record of your contribution (re-factor the PR yourself) or should I implement you changes myself?

cfoellmann commented:
One more thing:
Do not change the version number, please. Once ready it will be 2.0

Commit:
Adhere to WP coding standards + reset version number

- Put spaces on both sides of the opening and closing parenthesis of
if, elseif, foreach, for, and switch blocks.
- Rollback version number to 2.0

Commit:
Reset version number to 2.0

Commit:
Included previously removed DocBlocks

pryley commented:
I reverted the version number, added back the docblocks, and changed the spacing around if/else switch blocks.

I'm not sure what else (other than the spacing) goes against WP coding standards/code styling. My personal preference is to separate if/else/elseif blocks on their own lines, and use 'if(' instead of 'if (' for code alignment/readability. But I've changed it back.

[PHP Coding Standards](http://make.wordpress.org/core/handbook/coding-standards/php/)

Commit:
Adjustments

- change $this->set_glue_globals() back to public
- added an error message to die()

cfoellmann commented on wp-pydio-bridge.php:
```diff
@@ -197,27 +199,27 @@ public function admin_menu() {
-		</div>
-
-		<?php
+	public function options_page() 
```
Why did you make these changes?

pryley replied:
Oh, actually that was just personal preference. I try to avoid using PHP tags within PHP to output only small amounts of html. It also allows me to code-fold function blocks properly in Espresso.

Also:
1. With an empty action attribute the form was not submitting properly. `<form action='options.php' method='post'>`
2. `settings_fields( $option_group )` This should match the group name used in `register_setting()`

cfoellmann commented:
I needed to copy the improvements by hand. Sorry.
I stopped the merge process because it was nearly 100% change with ~50% "new" code.

I will move the auto/connection and wp-admin functionalities to separate classes. Maybe you can jump in there.

pryley commented:
Great! No problem.
